### PR TITLE
[21411] Table Header width too big in repository revision table

### PR DESCRIPTION
**Pull request by HDinger**
The table header has now the modifier class '-short', which removes the min-width for the element. 

https://community.openproject.org/work_packages/21411

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://hdinger-fix-21411-table-header-too-big-repository-revision-tabl.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://hdinger-fix-21411-table-header-too-big-repository-revision-tabl.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

**Comment by NobodysNightmare** [comment]
@ulferts Someone needs to explain this `-short` to me and @HDinger says it was all your idea ;-P

It seems quite arbitrary on which columns I will have to add this. Do we have to add it to all our tables now? Why doesn't the table know if a column is "short"?
I just tried removing the min width from all sort header rules, which seems to have the same effect.
But maybe I just did not try the right edge-case.

**Comment by basiszwo** [comment]
yeah, having to add the `-short` class sounds pretty error prone. the table should know by itself. I don't know the full reason why @ulferts introduced the `-short` class.

**Comment by ulferts** [comment]
We need to have a min-width on the table columns because they will otherwise become shortened to an unreadable extend:
image
The columns therefore receive a min-width of 150px. They then look like this:
image
Granted, the 150px is just guesswork but it is looking way better with the min-width than without it.

But there are columns, that we know to always be shorter. Those are e.g. columns having a checkbox (which have their own class called `checkbox` which should probably be renamed to `-short` for consistency). Those columns look weird if they have a min width of 150px:
image
But as we have a column width calculation for the column headers, we need to adapt those headers as well because the column header with the column highlight will otherwise reach into the next column header:
image

So my take on it is that we should introduce a -short class for the `td` elements as well and replace the `checkbox` class with it to have consistency.

**Comment by NobodysNightmare** [comment]
Just to make sure I understand the implications of the current approach.

We now have two kinds of columns: _short_ columns and _regular_ columns.

**short**:
- can be as small as they like
- if the title is longer than the column content, the title _may_ be truncated

**regular**:
- has a minimum width of 150px
- if the content is very small (e.g. checkboxes), **rendering errors will occur**
- if there is no content at all it works well for titles up to a width of 150px
- if there is no content at all, titles longer than 150px _may_ be truncated

### Example of a rendering error

This is the rendering error fixed by this PR:
render-error

**Comment by NobodysNightmare** [comment]
@ulferts I think I could live with most of the limitations of this approach and take it as "best we can get". But what really [grinds my gears](http://knowyourmeme.com/photos/559151-you-know-what-really-grinds-my-gears) is the fact that too short content will cause the bad rendering to happen.
I don't think that this is something we can control sufficiently to fix it using a static class per column.

Having a too wide column for short content or a truncated title after 150px for a populated table is IMO more acceptable than plainly inconsistent table rendering.

**Comment by ulferts** [comment]
@NobodysNightmare, @HDinger and myself talked about this. The problem here is the `id` class on the `td` element which has a max-width of "50px" defined.

**Comment by HDinger** [comment]
The actual version follows the idea of @ulferts and introduces a '-short' class for td-elements. Further all other widths declarations were removed. Because of this other tables than the revision table were changed now, too.

**Comment by basiszwo** [comment]
this is not fixed

bildschirmfoto 2015-09-18 um 00 08 46

**Comment by HDinger** [comment]
@basiszwo how did you test this? For me it works on every browser on big and small screen... Maybe you have to resize the table a bit because of the interactive table directive.

**Comment by basiszwo** [comment]
i checked out your code opened the repository page and took the screenshot.

##

**Comment by NobodysNightmare** [comment]
After a visit from @HDinger I verified this PR on my machine again and it seems to work.

@basiszwo Did you do the `assets:webpack` business? Or use foreman, who does take care for you? Maybe it has something to do with outdated stylesheets...

**Comment by basiszwo** [comment]
I use foreman. 
will recheck later.

File: app/assets/stylesheets/content/_table.sass [hunk 1]
```diff
@@ -139,23 +137,16 @@ table.generic-table
     td
       max-width: 300px
       min-width: 150px
-      overflow:  hidden
+      overflow:  hidden !important
```
Comment (ulferts) [inline app/assets/stylesheets/content/_table.sass]: @HDinger, that !important irks me. Can you get around this?

**Comment by ulferts** [comment]
@basiszwo any update from your side?

**Comment by basiszwo** [comment]
Can't reproduce on my side. so lgtm :+1:

**Pull request merged**